processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -750,6 +753,24 @@ private Assignment forgeWithElementMapping(Type sourceType, Type targetType, Sou
             return createForgedAssignment( source, methodRef, iterableMappingMethod );
         }
 
+        private Assignment forgeOptionalMapping(Type sourceType, Type targetType, SourceRHS source) {
+
+            targetType = targetType.withoutBounds();
+            ForgedMethod methodRef = prepareForgedMethod( sourceType, targetType, source, "?" );
```

Comment:
Using `?` here to represent an `Optional` type

Reply:
I have to admit, I'm not certain how to trigger an error that will print out this forge history. Would you have any guidance on the simplest way to do so? Once I can see what it looks like, I can add the comment.